plugins/hydra_ray_launcher/hydra_plugins/ray_launcher/conf/hydra/launcher/ray.yaml

```diff
@@ -0,0 +1,7 @@
+hydra:
+  launcher:
+    class: hydra_plugins.ray_launcher.RayLauncher
+ray:
```

Comment:
Not sure if this is what you meant by config stub @omry. For ray.init I feel like there are no obvious default values to include here for the user. Especially since ray automatically detects the number of cpus and gpus available to it.

Reply:
I mean params that can be going to the RayLauncher object.
About what to pass to ray.init(), even if defaults are automatic you can pass in the default values from the [ray API docs](https://ray.readthedocs.io/en/ray-0.6.0/api.html).

Something like:
```yaml
hydra:
  launcher:
    class: hydra_plugins.ray_launcher.RayLauncher
    params:
       ray:
         # The address of the Redis server to connect to. If this address is not provided, then this 
         # command will start Redis, a global scheduler, a local scheduler, 
         # a plasma store, a plasma manager, and some workers. It will also 
         # kill these processes when Python exits
         redis_address : null
         # Number of cpus the user wishes all local schedulers to be configured with (null auto detects)
         num_cpus : null   
         ...
```

To make things easier to configure, you can do:

```yaml
hydra:
  launcher:
    class: hydra_plugins.ray_launcher.RayLauncher
    params:
      ray: ${hydra.ray}

  ray:
    # The address of the Redis server to connect to. If this address is not provided, then this 
    # command will start Redis, a global scheduler, a local scheduler, 
    # a plasma store, a plasma manager, and some workers. It will also 
    # kill these processes when Python exits
    redis_address : null
    # Number of cpus the user wishes all local schedulers to be configured with (null auto detects)
    num_cpus : null   
    ...
```

The reason to put ray inside Hydra is that it will hide ray from the app config. this is a design choice.
now a user can override ray parameters with:
```
python foo.py hydra.ray.num_cpus=2
```

A user that looks at the Hydra config (--cfg hydra) will see all the ray options he/she can override (without the comments, for now).